### ISSUE TITLE
Character Global Filter for Recipe Listings

### ISSUE THREAD

**Issue by nightbasilisk** [Expugn/priconne-quest-helper#36]
**Is your feature request related to a problem? Please describe.**
The most useful thing for me when using this tool is looking up who uses an item and when. This is fine for things that only some characters use but when you look up what something like "Emerald Earrings" / エメラルドイヤリング a lot of characters use it and it just becomes a tad confusing to look at (3 rows of character icons on my screen). 

**Describe the solution you'd like**
I would like it if just how you have item box we have a "character box" where we select the characters we care for (since even if we have them we might not care to put anything into them any time soon) and based on what we have in the box it only shows those characters in the Character Usage lists. A "limit" on the rank would also be nice so rank 9 and so on dont show when in global we only have up to rank 7 (for example).

**Describe alternatives you've considered**
I have literally no idea how to solve this any other way.

**Additional context**
This is how it looks:
![image](https://user-images.githubusercontent.com/6383845/106413220-32f5d680-6452-11eb-9284-0a9ccb7a8119.png)
It's just too hard to parse that as "yes this other important character I have needs it too"

Ideally (with a global filter in place) it might look something more managable like this (in the worst case)
![image](https://user-images.githubusercontent.com/6383845/106413294-66d0fc00-6452-11eb-845f-0c90973adffc.png)

**If confusion is a problem** you can just make it so the global filter would just split the list into two lists. List 1 the things you put in your character box, list 2 everything else you didnt put in your character box (or doesnt meet your rank limit). That way there's no way anyone can shoot themselves in the foot with the filter.

**Comment by Expugn** [comment]
Your suggestion should be live, thanks!

Here's how I implemented it:
At the main page, the top left corner will have a "Character Box" button, clicking on it will reveal a page similar to this. I've selected all Kokkoro, Kyaru, and Pecorine units as an example
![image](https://user-images.githubusercontent.com/5342885/106432649-dd0a3a00-6423-11eb-812e-32b646c7973c.png)

Now when I preview information for `Emerald Earrings`, Character Usage should look like this
![image](https://user-images.githubusercontent.com/5342885/106432850-25c1f300-6424-11eb-90cd-33b0c05b45a2.png)

I didn't remove the other characters entirely in case the user would be interested in seeing characters they didn't have.

As for the rank filter, I don't think it's necessary because it's not like some `Rank 19` character would end up needing `Emerald Earrings` as a main item; sounds like more of a hassle for the user to keep updating their rank filter.

Having no characters in your character box will display Character Usage as it did initially.

Hopefully this is be a good enough solution to your suggestion.
